pkg/core/blockchain.go

```diff
@@ -330,8 +337,12 @@ func (bc *Blockchain) init() error {
 	if ver.Value != version {
 		return fmt.Errorf("storage version mismatch betweeen %s and %s", version, ver.Value)
 	}
-	bc.dao.StoragePrefix = ver.Prefix
-	bc.persistent.StoragePrefix = ver.Prefix // not strictly needed but we better be consistent here
+	if ver.StateRootInHeader != bc.config.StateRootInHeader {
```

Comment:
How about comparing dao's and config's `P2PSigExtensions`  values? They should be the same.